test/shell/test_coverage_scalatest.sh

```diff
@@ -4,15 +4,19 @@ dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 . "${dir}"/test_helper.sh
 runner=$(get_test_runner "${1:-local}")
 
+SCALA_VERSION="${SCALA_VERSION:-2.12.19}"
+
 test_coverage_on() {
-    bazel coverage //test/coverage_scalatest:test-scalatest
+    bazel coverage \
+        --repo_env="SCALA_VERSION=${SCALA_VERSION}" \
+        //test/coverage_scalatest:test-scalatest
     diff test/coverage_scalatest/expected-coverage.dat $(bazel info bazel-testlogs)/test/coverage_scalatest/test-scalatest/coverage.dat
 }
 
 test_coverage_includes_test_targets() {
     bazel coverage \
-          --instrument_test_targets=True \
-          //test/coverage_scalatest:test-scalatest
+        --instrument_test_targets=True \
```

Comment:
Shouldn't `--repo_env="SCALA_VERSION=${SCALA_VERSION}"` be passed here as well?

Reply:
Good catch! But I actually went the other way—the tests missing `--repo_env="SCALA_VERSION=${SCALA_VERSION}"` are only `grep`ping for the expected filename, not `diff`ing against an expected coverage file. So I've added comments in each file to explain the `SCALA_VERSION` setting.

I think the actual solution might be to generate coverage files for each Scala version, and use `$SCALA_VERSION` to select between them instead of setting `--repo_env`. I can take that on as a side quest, perhaps.